core/src/main/java/co/aikar/commands/CommandManager.java

```diff
@@ -92,4 +94,17 @@ public RegisteredCommand createRegisteredCommand(BaseCommand command, String cmd
     public abstract void log(final LogLevel level, final String message);
 
     public abstract void log(final LogLevel level, final String message, final Throwable throwable);
+
+    /**
+     * Registers an {@link ExceptionHandler} that is called when an exception occurs while executing a command.
+     *
+     * @param exceptionHandler the handler that should handle uncaught exceptions
+     */
+    public void registerExceptionHandler(ExceptionHandler exceptionHandler){
+        exceptionHandlers.add(exceptionHandler);
+    }
+
+    protected void handleUncaughtException(CommandIssuer sender, List<String> args, Exception e){
```

Comment:
This signature needs the BaseCommand that triggered the exception, and the RegisteredCommand.

Then the args should be the args for that subcommand.

And maybe Throwable instead of Exception?